packages/foreman/foreman-installer/foreman-installer.spec

```diff
@@ -66,72 +106,64 @@ rake install \
-	if [ -d $MIGRATIONS ] && [ ! -L $MIGRATIONS ] ; then
-		mv $MIGRATIONS/.applied %{_sysconfdir}/%{name}/scenarios.d/$scenario-migrations-applied
-		rm -rf $MIGRATIONS
-	fi
-done
-
-%files
+%post scenario-foreman-proxy-content
+foreman-installer --scenario foreman-proxy-content --migrations-only > /dev/null
+
+%files common
 %defattr(-,root,root,-)
 %doc README.*
 %license LICENSE
 %dir %{_sysconfdir}/%{name}
 %config(noreplace) %{_sysconfdir}/%{name}/custom-hiera.yaml
-%dir %{_sysconfdir}/%{name}/scenarios.d
-%{_sysconfdir}/%{name}/scenarios.d/foreman.migrations
-%config(noreplace) %attr(600, root, root) %{_sysconfdir}/%{name}/scenarios.d/foreman.yaml
-%config(noreplace) %attr(600, root, root) %{_sysconfdir}/%{name}/scenarios.d/foreman-answers.yaml
-%config(noreplace) %{_sysconfdir}/%{name}/scenarios.d/foreman-migrations-applied
+%dir %{scenariodir}
 %{_sbindir}/%{name}
-%{_datadir}/%{name}
-%{_mandir}/man8
-
-# katello files
-%exclude %{_datadir}/%{name}/config/foreman-proxy-content*
-%exclude %{_datadir}/%{name}/config/katello*
-%exclude %{_datadir}/%{name}/katello-certs
-%exclude %{_datadir}/%{name}/parser_cache/foreman-proxy-certs.yaml
-%exclude %{_datadir}/%{name}/parser_cache/foreman-proxy-content.yaml
-%exclude %{_datadir}/%{name}/parser_cache/katello.yaml
-
-%files katello
-# common
+%dir %{_datadir}/%{name}
+%{_datadir}/%{name}/checks
+%dir %{_datadir}/%{name}/config
+%{_datadir}/%{name}/config/config_header.txt
+%{_datadir}/%{name}/config/foreman-hiera.yaml
+%{_datadir}/%{name}/config/foreman.hiera
+%{_datadir}/%{name}/hooks
+%{_datadir}/%{name}/modules
+%{_datadir}/%{name}/VERSION
+%dir %{parser_cache}
+%{_mandir}/man8/%{name}.8*
+
+%files katello-common
 %{_sbindir}/katello-certs-check
 
-# foreman-proxy-content scenario
-%{_datadir}/%{name}/config/foreman-proxy-content*
-%{_datadir}/%{name}/parser_cache/foreman-proxy-content.yaml
-%{_sysconfdir}/%{name}/scenarios.d/foreman-proxy-content.migrations
-%config(noreplace) %attr(600, root, root) %{_sysconfdir}/%{name}/scenarios.d/foreman-proxy-content.yaml
-%config(noreplace) %attr(600, root, root) %{_sysconfdir}/%{name}/scenarios.d/foreman-proxy-content-answers.yaml
-%config(noreplace) %{_sysconfdir}/%{name}/scenarios.d/foreman-proxy-content-migrations-applied
-
-# katello scenario
-%{_datadir}/%{name}/config/katello*
-%{_datadir}/%{name}/parser_cache/katello.yaml
-%{_sysconfdir}/%{name}/scenarios.d/katello.migrations
-%config(noreplace) %attr(600, root, root) %{_sysconfdir}/%{name}/scenarios.d/katello.yaml
-%config(noreplace) %attr(600, root, root) %{_sysconfdir}/%{name}/scenarios.d/katello-answers.yaml
-%config(noreplace) %{_sysconfdir}/%{name}/scenarios.d/katello-migrations-applied
+%files scenario-foreman
+%{configdir}/foreman.migrations
+%config(noreplace) %attr(600, root, root) %{scenariodir}/foreman.yaml
+%config(noreplace) %attr(600, root, root) %{scenariodir}/foreman-answers.yaml
+%config(noreplace) %{scenariodir}/foreman-migrations-applied
+%{parser_cache}/foreman.yaml
```

Comment:
this could become a macro `files_for_scenario(name)`, but not sure if that'd be over-optimization or not

Reply:
I tried to write a macro but couldn't make it work. There's a good chance that's on my side but RPM documentation was insufficient.